src/state/optionsSlice.ts

```diff
@@ -1,13 +1,29 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { TooltipEventType } from '../util/types';
+import { TooltipPayloadSearcher } from './tooltipSlice';
 
 export type ChartOptions = {
   chartName?: string;
   defaultTooltipEventType?: TooltipEventType;
   validateTooltipEventTypes?: ReadonlyArray<TooltipEventType>;
+  // Should this instead be a property of a graphical item? Do we want to mix items with different data types in one chart?
+  tooltipPayloadSearcher: TooltipPayloadSearcher | undefined;
 };
 
-const initialState: ChartOptions = {};
+export const arrayTooltipSearcher: TooltipPayloadSearcher = (
+  data: unknown[],
+  strIndex: string | undefined,
+): unknown => {
+  const numIndex = Number(strIndex);
```

Comment:
Treemap data are nested, there's no straightforward numerical index I can think of.

Reply:
Yeah ack. Unless we didn't need indices and everything was provided as you hover/click/etc. but idt that can become a thing while all transformations happen outside of the chart

Reply:
Hm on the other hand if we ever want to support keyboard controls for tooltip, we will need that one-dimensional array _anyway_. So it might have been just easiest to do that now.

Oh well. It will be easy to refactor in the future unless we expose this as an external API.